src/abe_policy/policy.rs

```diff
@@ -156,8 +156,7 @@ impl Policy {
         self.dimensions
             .iter()
             .flat_map(|(dim_name, dim)| {
-                dim.attributes
-                    .keys()
+                dim.get_attributes_name()
```

Comment:
Maybe rename this to `keys()` to follow the `HashMap` conventions.
```suggestion
                dim.keys()
```

Reply:
`Dimension` is an abstraction so I am not sure it's a good idea to use the same names as the underlying structs (which could change)